Support PHP 8.5

Jan-Schuppik opened the pull request:
## Changes made:
### PHP 8.4:
- [Function parameters that are null by default must be declared nullable](https://www.php.net/manual/en/migration84.deprecated.php#migration84.deprecated.core).

### PHP 8.5:
- [Using null as an array offset is deprecated.](https://www.php.net/manual/en/migration85.deprecated.php#migration85.deprecated.core.using-null-as-an-array-offset)
- Remove deprecated method [`ReflectionProperty::setAccesible()`](https://www.php.net/manual/en/migration85.deprecated.php#migration85.deprecated.reflection)

### Other changes:
- Update composer package `guzzlehttp/psr7`

resolves #181

Jan-Schuppik committed:
PHP 8.4: Change implicit nullable type declaration to explicit

Since PHP 8.4 implicitly nullable parameter types are deprecated.

Jan-Schuppik committed:
Update composer packages `guzzlehttp/psr7`

Jan-Schuppik committed:
PHP 8.5: Remove deprecated method `ReflectionProperty::setAccessible()`

Since we require PHP >= 8.2 and since calling this method has no effect as of
PHP 8.1.0, as all properties are accessible by default, it is safe to simply
remove this method call.

sukhwinder33445 reviewed (dismissed):
Looks good

lippserd commented on src/FormElement/RadioElement.php:
```diff
@@ -68,6 +68,9 @@ public function setOptions(array $options): self
     {
+        // php>=8.5 does not support null as array key
+        $value = $value ?? '';
+
```
The following question applies to all other code locations that you have introduced in this way: Why is this necessary? Because PHPDoc does not allow `null`.

lippserd commented on tests/FormElement/RadioElementTest.php:
```diff
@@ -282,6 +283,7 @@ public function testRadioNotValidIfCheckedValueIsDisabled()
         $this->assertFalse($radio->isValid());
     }
 
+    #[RequiresPhp('< 8.5')]
```
Since using `null` as an array offset is deprecated and should be discouraged, I wouldn't even show anymore that we support this for PHP < 8.5. I suggest removing all test cases where `null` is used as an array offset. Please pay special attention to tests where the empty string and `null` are tested differently, e.g., for `value`.

lippserd reviewed (changes requested):
In addition to the requested changes, I have a few comments on the commit messages and descriptions:

For 982ef2846935678e133983d82059114cf827bffe, please use something like the following, as suggested in https://github.com/Icinga/ipl-stdlib/pull/61#pullrequestreview-3592029375:

```
PHP 8.4: Change implicit nullable type declaration to explicit

Since PHP 8.4 implicitly nullable parameter types are deprecated.
```

The reason for upgrading `guzzlehttp/psr7` in d5b09891733a7004724259c1c3864dfc1d34f789 is misleading, as version 2.8.0 introduces PHP 8.5 support and is a dependency for this library. In general, our ipl-* repositories build our Icinga PHP library, not the other way around. Therefore, we should never have such "streamline" commits. Something similar was also discussed/addressed in https://github.com/Icinga/ipl-validator/pull/38.

For 41413f04793a8c9fc9d7a2bd65bc64e2ec678b11, the commit message could be worded more precisely, and a description justifying the removal should be added, e.g.:

```
PHP 8.5: Remove deprecated method `ReflectionProperty::setAccessible()`

Since we require PHP >= 8.2 and since calling this method has no effect as of
PHP 8.1.0, as all properties are accessible by default, it is safe to simply
remove this method call.
```

Finally, 3ef2e009b2879316d6effb2b78a2d88d5d95bdeb could be rephrased to ```PHP 8.5: Replace usage of `null` as array offset```. A description that only contains a reference to the PHP documentation is not necessary, but something like the following:

```
Since PHP 8.5 using `null` as an array offset is deprecated. Code relying on 
`null` being silently converted to an empty string must be replaced and
discouraged in PHPDocs.
```

lippserd commented:
And please also add a PR description.

sukhwinder33445 committed:
PHP 8.5: Replace usage of `null` as array offset

Since PHP 8.5 using `null` as an array offset is deprecated. Code relying on
`null` being silently converted to an empty string must be replaced and
discouraged in PHPDocs.

sukhwinder33445 committed:
`(Radio|Select)Element::getOption()`: Remove support for `null` as param

- (Radio|Select)Option: Remove support for `null` value by defining the parameter types for the constructor
- Ensure empty-string `''` option cannot be `disabled|checked|selected` by falsy values except `null` and `''`
- Adjust tests accordingly and remove redundant tests

lippserd reviewed (changes requested):
Both concerns regarding the radio element also apply to the select element.

lippserd commented on src/FormElement/RadioElement.php:
```diff
@@ -44,10 +44,7 @@ public function setOptions(array $options): self
-                    in_array($value, $this->disabledOptions, ! is_int($value))
-                    || ($value === '' && in_array(null, $this->disabledOptions, true))
-                );
+                ->setDisabled(in_array($value, $this->disabledOptions));
```
I'm sorry, I misread the code. Other types were also applied previously, only the empty string was handled specially. Without your changes, the code would result in the following:

```php
var_dump($radio->getOption('')->isDisabled()); // bool(false)
var_dump($radio->getOption(0)->isDisabled()); // bool(true)
var_dump($radio->getOption(1)->isDisabled()); // bool(true)
```

Of course, it still makes sense to fix this. And I would introduce tests for the other types so that we don't accidentally change the behavior.

sukhwinder33445 committed:
`(Radio|Select)ElementTest`: Extend tests

lippserd reviewed (changes requested):
Since explicit is better than implicit, I also suggest adding new `assertNull()` calls after the `setValue()` calls at the end of `testNullAndTheEmptyStringValuesAreEquallyHandled()` for both `Radio` and `SelectElementTest`. If I'm not mistaken, there is no check to validate if `setValue(''|null)` leads to `getValue(): null`.

lippserd reviewed (approved):
Thanks 👍
